### PR TITLE
WT-13324 Cleanup test_compact07

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: test/suite/test_compact07.py [hunk 1]
```diff
@@ -104,24 +104,24 @@ def test_compact07(self):
...
         # table created but only the others with more empty space.
-        self.session.compact(None, f'background=true,free_space_target={free_space_20 + 1}MB')
-
-        # Wait for the background server to wake up.
-        compact_running = self.get_bg_compaction_running()
-        while not compact_running:
-            time.sleep(1)
-            compact_running = self.get_bg_compaction_running()
-        self.assertEqual(compact_running, 1)
+        assert min_free_space - 1 > free_space_20
```
Comment (mm-ng) [inline test/suite/test_compact07.py:118]: Should we use `self.assertGreater` here rather than a plain `assert`?

File: test/suite/test_compact07.py [hunk 2]
```diff
@@ -104,24 +104,24 @@ def test_compact07(self):
...
         # There should be more free space in the last created tables compared to the very first one.
+        min_free_space = 0
         for i in range(self.n_tables):
-            self.assertGreater(self.get_free_space(uri), free_space_20)
+            free_space = self.get_free_space(uri)
+            self.assertGreater(free_space, free_space_20)
+            if i == 0:
+                min_free_space = free_space
+            elif free_space < min_free_space:
+                min_free_space = free_space
```
Comment (Sean04) [inline test/suite/test_compact07.py:114]: Why is this necessary? The space we're deleting is hard coded.

File: test/suite/test_compact07.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/test_compact07.py]: Sure b5259ec22c73c06c326b3ef782841eb2980378f4

File: test/suite/test_compact07.py (hunk 2 above)
Comment (etienneptl) [reply test/suite/test_compact07.py:114]: That's true, I was being overly cautious in case the free space is not as much as it should have been.

Comment (etienneptl) [reply test/suite/test_compact07.py]: 2f72748230f28d1bdf9076116a3b34415bad4d96

File: test/suite/test_compact07.py [hunk 3]
```diff
@@ -143,9 +136,6 @@ def test_compact07(self):
         # Check that foreground compaction has done some work on the small table.
         self.assertGreater(self.get_pages_rewritten(uri_small), 0)
 
-        # Check that we have some files in the background compaction tracking list.
```
Comment (etienneptl) [inline test/suite/test_compact07.py:146]: I think we should remove this as the stat may be back to 0 by now if background compaction has compacted everything and there is no more work to do by the time we get here?

Comment (etienneptl) [reply test/suite/test_compact07.py:146]: Or I could move it earlier, the debug idle time is actually 10 seconds:
```
WT_BACKGROUND_COMPACT_MAX_IDLE_TIME_DEBUG
```

Comment (Sean04) [reply test/suite/test_compact07.py:146]: happy to remove it, it's flakey with timing.

**Review by Sean04 — APPROVED** [review]
lgtm

File: test/suite/test_compact07.py (hunk 3 above)
Comment (etienneptl) [reply test/suite/test_compact07.py:146]: I moved it to earlier in the test
8fea41e78b08af9cb435e5234c3d9b5a2c5abc9b

**Review by mm-ng — APPROVED** [review]
LGTM
